src/ceph-daemon

```diff
@@ -0,0 +1,1289 @@
+#!/usr/bin/env python3
+
+DEFAULT_IMAGE='ceph/daemon-base'
+DATA_DIR='/var/lib/ceph'
+LOG_DIR='/var/log/ceph'
+UNIT_DIR='/etc/systemd/system'
+VERSION='unknown development version'
+
+PODMAN_PREFERENCE = ['podman', 'docker']  # prefer podman to docker
+
+"""
+You can invoke ceph-daemon in two ways:
+
+1. The normal way, at the command line.
+
+2. By piping the script to the python3 binary.  In this latter case, you should
+   prepend one or more lines to the beginning of the script.
+
+   For arguments,
+
+       injected_argv = [...]
+
+   e.g.,
+
+       injected_argv = ['ls']
+
+   For reading stdin from the '--config-and-json -' argument,
+
+       injected_stdin = '...'
+"""
+
+import argparse
+import configparser
+import json
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import time
+from distutils.spawn import find_executable
+
+try:
+    from StringIO import StringIO
+except:
+    pass
+try:
+    from io import StringIO
+except:
+    pass
+
+logging.basicConfig(level=logging.INFO)
+
+podman_path = None
+
+##################################
+
+def pathify(p):
+    if '/' not in p:
+        return './' + p
+    return p
+
+def get_hostname():
+    import socket
+    return socket.gethostname()
+
+def make_fsid():
+    import uuid
+    return str(uuid.uuid1())
+
+def is_fsid(s):
+    import uuid
+    try:
+        uuid.UUID(s)
+    except:
+        return False
+    return True
+
+def makedirs(dir):
+    os.makedirs(dir, exist_ok=True)
+
+def find_program(filename):
+    name = find_executable(filename)
+    if name is None:
+        raise ValueError(f'{filename} not found')
+    return name
+
+def get_data_dir(base, fsid, t, n):
+    return base + '/' + fsid + '/' + t + '.' + n
+
+def get_log_dir(base, fsid):
+    return base + '/' + fsid
+
+def get_unit_name(fsid, daemon_type, daemon_id):
+    return 'ceph-%s@%s.%s' % (fsid, daemon_type, daemon_id)
+
+def check_unit(unit_name):
+    try:
+        out = subprocess.check_output(['systemctl', 'is-enabled', unit_name])
+        enabled = out.decode('utf-8').strip() == 'enabled'
+    except:
```

Comment:
this needs to catch at least `except Exception:` and eating it out without any log output of what the problem was is going to make it impossible to improve or fix if there are issues